src/wasm-interpreter.h

```diff
@@ -2228,6 +2228,186 @@ class ModuleInstance
     : ModuleInstanceBase(wasm, externalInterface) {}
 };
 
+static const Name NONSTANDALONE_FLOW("Binaryen|nonstandalone");
+
+typedef std::unordered_map<LocalGet*, Literals> GetValues;
+
+// Evaluates a standalone expression. Errors if we hit anything that can't be
+// evaluated.
+class StandaloneExpressionRunner
```

Comment:
I'm not sure about the name here. We evaluate a single expression, yes, but it isn't "standalone" in the sense that we have local and global values, and the globals in particular are from the "outside" in some sense...

What is mainly added here are local and global values, but I can't think of a good name for that atm...

Reply:
Conceptionally it evaluates an expression "out of context of its function", except if one gives it context by specifying `getValues` or explicitly setting local or global values, while still assuming some global context (most notably immutable globals). Perhaps `GeneralizedExpressionRunner`?